docs/migrating-from-0.9.md

```diff
@@ -0,0 +1,251 @@
+# Migrating from 0.9
+
+Genkit 1.0 introduces some of breaking changes alongside feature enhancements that improve overall functionality. If you have been developing applications with Genkit 0.9, you will need to update your application code when you upgrade to the latest version. This guide outlines the most significant changes and offers steps to migrate your existing applications smoothly.
+If you're using 0.5, use <> to upgrade to 0.9 first before upgrading to 1.0
+
+
+## Streaming functions do not require an await
+
+
+**Old:**
+
+```ts
+const { stream, response } = await ai.generateStream(`hi`);
+const { stream, output } = await myflow.stream(`hi`);
+```
+
+**New**
+
+```ts
+const { stream, response } = ai.generateStream(`hi`);
+const { stream, output } = myflow.stream(`hi`);
+```
+
+
+## Beta APIs
+
+We're introducing an unstable, Beta API channel, and leaving session, chat and Genkit client APIs in beta as we continue to refine them. 
+
+Note: When using the APIs as part of the Beta API, you may experience breaking changes outside of SemVer. Breaking chanhes may occur on minor release (we'll try to avoid making breaking beta API changes on patch releases).
+
+**Old:**
+
+```ts
+import { genkit } from 'genkit';
+const ai = genkit({...})
+const session = ai.createSession({ ... })
+```
+
+**New**
+
+```ts
+import { genkit } from 'genkit/beta';
+const ai = genkit({...})
+const session = ai.createSession({ ... })
+```
+
+**Old:**
+
+```ts
+import { runFlow, streamFlow } from 'genkit/client';
+```
+
+**New**
+
+```ts
+import { runFlow, streamFlow } from 'genkit/beta/client';
+```
+
+## Introducing new `@genkit-ai/express` package
+
+The package contains utilities to make it easier to build an express app using Express.js.
+
+Refer to https://js.api.genkit.dev/modules/_genkit-ai_express.html for more details.
+
+`startFlowServer` has moved from part of the genkit object to the express package, to use startFlowServer, update your imports. 
+
+
+**Old:**
+
+```ts
+const ai = genkit({ ... });
+ai.startFlowServer({
+  flows: [myFlow1, myFlow2],
+});
+```
+
+**New**
+
+```ts
+import { startFlowServer } from '@genkit-ai/express';
+startFlowServer({
+  flows: [myFlow1, myFlow2],
+});
+```
+
+## Changes to Flows
+
+`run` function for custom trace blocks has moving to `ai.run`
+
+**Old:**
+
+```ts
+ai.defineFlow({name: 'banana'}, async (input, streamingCallback) => {
+  const step = await run('myCode', async () => {
+    return 'something'
+  });
+})
+```ts
+
+**New**
+
+```ts
+ai.defineFlow({name: 'banana'}, async (input, {context, sendChunk}) => {
+  const step = await ai.run('myCode', async () => {
+    return 'something'
+  });
+})
+```
+
+`ai.defineStreamingFlow` has been removed. Use `ai.defineFlow` instead. Also, `streamingCallback` moved to a field inside the second argument of he flow function and is now called `sendChunk`.
```

Comment:
The second section move above

Reply:
will keep it here... I removed streaming callback above